operator/src/main/java/oracle/kubernetes/operator/http/HttpClient.java

```diff
@@ -196,11 +196,15 @@ public NextAction apply(Packet packet) {
       if (secretData != null) {
         username = secretData.get(SecretHelper.ADMIN_SERVER_CREDENTIALS_USERNAME);
         password = secretData.get(SecretHelper.ADMIN_SERVER_CREDENTIALS_PASSWORD);
-      }
-      packet.put(KEY, createAuthenticatedClient(username, password));
+        packet.put(KEY, createAuthenticatedClient(username, password));
 
-      Arrays.fill(username, (byte) 0);
-      Arrays.fill(password, (byte) 0);
+        if (username != null) {
```

Comment:
I see that you are addressing a potential bug here; it would be better, though, if you took the opportunity to clarify what is going on. For example. You could create a method:
```
// Erase authentication credential so that it is not sitting in memory where a rogue program can find it.
private void clearCredential(byte[] credential) {
  if (credential != null) Arrays.fill(credential, (byte) 0);
}
```
and then use that method for both variables. That would not only eliminate duplication, but would also make it clear why we're doing it.